scripts/conda-install.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# This script installs OpenMC, its Python API, and all necessary dependencies
+# into a conda environment. Because it uses compilers and CMake from the
+# Anaconda repository, it's not necessary to have those installed on your
+# system. However, at a minimum, you do need to have 'make' installed as well as
+# a linker. To get the OpenMC git repository, you'll need git installed as
+# well. On Debian derivatives, you can install all of these with:
+#
+# sudo apt install -y make binutils git
+
+set -e
+
+#===============================================================================
+# INSTALLATION OPTIONS
+
+INSTALL_MINICONDA=yes                             # Install miniconda?
+INSTALL_IN_CONDA=yes                              # Install OpenMC executable/package directly in conda environment?
+INSTALL_EDITABLE=no                               # Install in editable mode (overrides INSTALL_IN_CONDA)
+INSTALL_PREFIX=$HOME/opt/openmc/$(date +%Y-%m-%d) # Where to install (if INSTALL_IN_CONDA and INSTALL_EDITABLE are disabled)
+WITH_MPI=yes                                      # Install with MPI support
+PYTHON_VERSION=3.7                                # What version of Python to use
+ENVNAME=openmc-py$PYTHON_VERSION                  # conda environment name
+
+#===============================================================================
+
+# Download and install miniconda
+if [[ $INSTALL_MINICONDA == yes ]]; then
+    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+    bash Miniconda3-latest-Linux-x86_64.sh -b
+    $HOME/miniconda3/bin/conda update -y conda
+    $HOME/miniconda3/bin/conda init bash
+fi
+
+# Make sure conda is activated
+. $HOME/miniconda3/etc/profile.d/conda.sh
+conda activate
+
+# Create new Python environment to install everything into
+conda create -y -n $ENVNAME python=$PYTHON_VERSION
+conda activate $ENVNAME
+
+# Install recent version of gcc and CMake. OpenMC requires gcc 4.9+ and CMake
+# 3.3+, so if you are on an older distribution, it's possible that your version
+# is not sufficient.
+conda install -y gcc_linux-64 gxx_linux-64 gfortran_linux-64 cmake
+
+# Install OpenMC's Python dependencies
+conda install -y numpy scipy cython h5py pandas lxml matplotlib ipython pytest
```

Comment:
A minor issue here is `h5py` is a serial version by default, which will prevent any mpi-running of openmc in depletion or other similar capi-based simulations which will write out hdf5 results. So you may consider [installing parallel h5py using pip](http://docs.h5py.org/en/stable/build.html#building-against-parallel-hdf5):
``` bash
if [[ $WITH_MPI == yes ]]; then
    export HDF5_MPI="ON" 
    export HDF5_DIR=/usr/lib/x86_64-linux-gnu/hdf5/mpich
    pip install --no-binary=h5py h5py
fi
```

Reply:
Good point. I just fixed this, although slightly different than what you suggested (using parallel h5py from conda-forge).